proto/gz/msgs/cameratrack.proto

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2024 Rudis Laboratories
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package gz.msgs;
+option java_package = "com.gz.msgs";
+option java_outer_classname = "CameraTrackProtos";
+
+/// \ingroup gz.msgs
+/// \interface CameraTrack
+/// \brief Message for camera tracking
+
+import "gz/msgs/entity.proto";
+import "gz/msgs/header.proto";
+import "gz/msgs/vector3d.proto";
+
+message CameraTrack
+{
+  enum TrackMode
+  {
+    /// \brief Defaults to no tracking mode.
+    NONE              = 0;
+    /// \brief TRACK observes a target from fixed x,y,z world 
+    ///   camera point with pgain allowing only for orientation change.
+    TRACK             = 1;
+    /// \brief FOLLOW moves with a target with an offset x,y,z camera and pgain.
+    FOLLOW            = 2;
+    /// \brief FOLLOW_FREE_LOOK moves with a target with an offset x,y,z camera
+    ///   and pgain but allows for changing orientation for free looking around.
+    FOLLOW_FREE_LOOK  = 3;
+    /// \brief FOLLOW_LOOK_AT moves with a target with an offset x,y,z camera
+    ///   and pgain while looking at a specified target.
+    FOLLOW_LOOK_AT    = 4;
+    /// \brief USE_LAST updates offsets and pgains without changing the
+    ///   fundamental track_mode or target(s)
+    USE_LAST          = 5;
+  }
+
+  /// \brief Optional header data
+  Header header           = 1;
+  TrackMode track_mode    = 2;
+  Entity follow_target    = 3;
+  Entity track_target     = 4;
+  Vector3d follow_offset  = 5;
+  Vector3d track_offset   = 6;
+  double follow_pgain     = 7;
+  double track_pgain      = 8;
```

Comment:
Could you please document these? 

From my understanding, the mode is (essentially) either track or follow. When is the case where both `follow` and `track` variables are required? I'm wondering if there should only be 1 variable for `mode`, `offset`, and `pgain` instead?

Reply:
So, they actually are differentiable where you can have different pgains for both, same thing for offset and modes is needed to enable all those new types of "follow" and "track". And actually default "follow" implements "follow" and "track" where they are the same target.

Reply:
I see, thanks for the explanation. When is there a case when `follow_target` and `track_target` are the same or different? Wouldn't it be cleaner to specify the `track_mode` then set the `target`, `offset`, and `pgain` accordingly what type of mode you're in?

Reply:
I think this was covered on the call, but it's pretty much any mode other than track and follow_free_look that uses both track and follow.